### PR TITLE
support persistables in v2 tables

### PR DESCRIPTION
**Pull request by nziebart**
**Goals (and why)**:
Support persistables in v2 generated code

Fixes https://github.com/palantir/atlasdb/issues/2485

**Implementation Description (bullets)**:
- Expose a `getJavaObjectTypeClass` method on `ColumnValue`, which correctly returns the underlying class
- Add a column to the test schema containing a persistable

TODO:
- add docs indicating this feature is beta

**Concerns (what feedback would you like?)**:

**Where should we start reviewing?**:

**Priority (whenever / two weeks / yesterday)**:
this week

---
This change is [Reviewable](https://reviewable.io/reviews/palantir/atlasdb/2495)

### CODE REVIEW

**Comment by nziebart** [comment]
TODO: we also want a test for multiplex tow components, but will do this in a follow up PR

File: atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestTable.java [hunk 1]
```diff
@@ -533,25 +533,25 @@ public void putColumn1UnlessExists(Map<SchemaApiTestRow, Long> map) {
         putUnlessExists(Multimaps.forMap(toPut));
     }
 
-    public void putColumn2(SchemaApiTestRow row, String value) {
+    public void putColumn2(SchemaApiTestRow row, com.palantir.atlasdb.table.description.StringValue value) {
         put(ImmutableMultimap.of(row, Column2.of(value)));
     }
 
-    public void putColumn2(Map<SchemaApiTestRow, String> map) {
+    public void putColumn2(Map<SchemaApiTestRow, com.palantir.atlasdb.table.description.StringValue> map) {
```
Comment (nziebart) [inline atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestTable.java]: hmm, need to get rid of these package qualifiers

Comment (nziebart) [reply atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestTable.java]: actually i don't think this behavior changed for the v1 generated code, so leaving as is

File: atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java [hunk 2]
```diff
@@ -166,7 +167,7 @@ public Namespace getNamespace() {
 
     /**
      * Returns the value for column Column2 and specified row components. */
-    public Optional<String> getColumn2(String component1) {
+    public Optional<StringValue> getColumn2(String component1) {
```
Comment (hsaraogi) [inline atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java]: This does not compile for me:

```
/Users/hsaraogi/code/atlasdb/atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java:330: error: cannot find symbol
            StringValue newValue = processor.apply(result.get());
            ^
  symbol:   class StringValue
  location: class SchemaApiTestV2Table
33 errors
:atlasdb-client:compileIntegrationInputJava FAILED
```

Comment (nziebart) [reply atlasdb-client/src/integrationInput/java/com/palantir/atlasdb/table/description/generated/SchemaApiTestV2Table.java]: yep, not compiling in circle either. let me fix it

**Comment by fsamuel-bs** [comment]
This should be good for review again. Pinging @hsaraogi for it.

**Comment by hsaraogi** [comment]
Reviewed 3 of 14 files at r1, 2 of 2 files at r2, 10 of 12 files at r4, 2 of 3 files at r5, 1 of 1 files at r6.
Review status: all files reviewed at latest revision, 2 unresolved discussions, some commit checks failed.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/palantir/atlasdb/2495)*

File: docs/source/release_notes/release-notes.rst [hunk 3]
```diff
@@ -111,10 +115,19 @@ develop
           This will provide more accurate metrics for when sweep is doing something.  Sweeping run through the sweep endpoint will now also contribute to these metrics, before it didn't update any metrics which again distorted the view of what work sweep was doing on the DB.
           (`Pull Request <https://github.com/palantir/atlasdb/pull/2535>`__)
 
-    *   - |devbreak|
-        - Simplify and annotate the constructors for ``SerializableTransactionManager``. This should make the code of that class more maintainable.
-          If you used one of the deleted or deprecated constructors, use the static ``create`` method.
-          (`Pull Request <https://github.com/palantir/atlasdb/pull/2549>`__)
+    *    - |improved| |devbreak|
```
Comment (hsaraogi) [inline docs/source/release_notes/release-notes.rst]: Looks like the merge conflict wasnt resolved in the right manner, this PR is effectively adding two release note entries.

**Pull request merged**